### PR TITLE
Update DG use cases for v1.1

### LINKED ISSUE #14: Update DG - Use Case
Provide use cases (textual form) for a few representative user stories that need multiple steps to complete. Eg: Steps to add a tag to a user. #12

### PR DESCRIPTION
**Pull request by RuiFengg**
Use cases added for modules and tasks

Task: add, delete, update, presets
Module: add, delete, presets

### CODE REVIEW

**Comment by RuiFengg** [comment]
Better to merge this after everyone updates their profile in the aboutus page

**Review by khoongwk — COMMENTED** [review]
Our profiles are done, shall we merge this PR into master?

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -266,31 +266,196 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**Guarantees: List of all modules available**
+
+**MSS**
+
+1. User request to list all modules
+2. GrAB3 shows a list of all modules
+Use case ends
+
+**UC09: View MyModules**
```
Comment (hanstirtaputra) [inline docs/DeveloperGuide.md]: I think we are going to omit MyModules from the architecture, we will be using ModuleSchedule instead, so this use case should be _View ModuleSchedule_ instead

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -266,31 +266,154 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User requests to list all tasks for the specified week.
+2.  GrAB3 displays a list of tasks for that week.
+3.  User requests to delete a specific task from the list generated.
+4.  GrAB3 shows the updated list after deletion of the task.
 
     Use case ends.
 
 **Extensions**
```
Comment (khoongwk) [inline docs/DeveloperGuide.md:305]: Should we add in a case for invalid week specified by the user as an extension to [1]? Like week 123 or week -1. Same thing for UC03 - editing a task.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -266,31 +266,154 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**Preconditions: Module list must exist**
+
+**Guarantees: A new module will be added to the module schedule**
+
+**MSS**
+
+1.  User requests to view all modules available.
+2.  GrAB3 shows the module list.
+3.  User selects module to add into module schedule
```
Comment (khoongwk) [inline docs/DeveloperGuide.md]: Perhaps we should add an extension for the case when a user inputs a module code that's not in the list? Like what you did for UC07.

**Review by khoongwk — CHANGES_REQUESTED** [review]
I think a couple of minor edits are needed to account for cases where user enters a wrong input. Other than that, LGTM!

**Comment by RuiFengg** [comment]
Amend commit to update DG use cases with suggestions from @khoongwk

**Review by hanstirtaputra — APPROVED** [review]
Looks good as a draft! But I think we will need to add more details regarding the target user profile to really narrow down our target audience.

**Review by khoongwk — APPROVED** [review]

**Pull request merged**